### PR TITLE
add: blog app

### PR DESCRIPTION
**Pull request by alvinisasi**

### CODE REVIEW

File: apps/blog/pages/about/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export const getStaticProps: GetStaticProps = async () => {
+    const queryClient = new QueryClient()
+  
+    await queryClient.prefetchQuery(queries.about.all())
+  
+    return {
+        props: {
+            dehydratedState: dehydrate(queryClient),
+        },
+        revalidate: revalidate
```
Comment (exneval) [inline apps/blog/pages/about/index.tsx:64]: `revalidate: revalidate` -> `revalidate`

File: apps/blog/pages/about/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+export const revalidate = 60
+
+const About = ({ dehydratedState }: InferGetStaticPropsType<typeof getStaticProps>) => {
+    const theme = useTheme()
+    const { data, isLoading } = useQuery(queries.about.all())
+    
+    if(data){
+        const attributes = data[0].attributes
+        return(
+            <HydrationBoundary state={dehydratedState}>
```
Comment (exneval) [inline apps/blog/pages/about/index.tsx:17]: should wrap with hydration context in _app file instead

File: apps/blog/pages/about/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                                {parse(attributes.content)}
+                            </Typography>
+                        </Container>
+                } 
+                    
+                </Box>
+            </HydrationBoundary>
+        )
+    }
+    return <Typography variant="body1" color={theme.palette.primary.contrastText}>
```
Comment (exneval) [inline apps/blog/pages/about/index.tsx:49]: no prettier formatting applied

File: apps/blog/pages/posts/[slug]/index.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import { IParams, PostResponse } from "@/utils/types"
+import { Box, Container, Typography, useTheme } from "@mui/material"
+import DateRangeIcon from '@mui/icons-material/DateRange';
+import { getDate } from "@/utils/helper";
+import Image from "next/image";
+import parse from 'html-react-parser';
+import { HydrationBoundary, QueryClient, dehydrate, useQuery } from "@tanstack/react-query";
+import { getPosts } from "@/services/posts";
+import { GetStaticProps, GetStaticPaths, InferGetStaticPropsType  } from 'next';
+import { usePathname } from 'next/navigation'
```
Comment (exneval) [inline apps/blog/pages/posts/[slug]/index.tsx:10]: get the pathname from next/router hooks instead

File: apps/blog/pages/posts/[slug]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+const PostDetail = ({ dehydratedPost }: InferGetStaticPropsType<typeof getStaticProps>) => {
+    const theme = useTheme()
+    const pathname = usePathname()
+    const { data, isError, isLoading } = useQuery(queries.posts.detail(pathname?.replace('/posts/', '') || ''))
+    let arr = [0, 1, 2, 3, 4]
+
+    if(data && !isError){
+        const attributes =  data[0].attributes
+        return(
+            <HydrationBoundary state={dehydratedPost}>
```
Comment (exneval) [inline apps/blog/pages/posts/[slug]/index.tsx:24]: should wrap with hydration context in _app file

File: apps/web/app/layout.tsx
Comment (exneval) [inline apps/web/app/layout.tsx:1]: use only pages dir to avoid confusing

File: apps/web/app/globals.css
Comment (exneval) [inline apps/web/app/globals.css:1]: only use pages dir to avoid confusing

File: apps/web/app/favicon.ico
Comment (exneval) [inline apps/web/app/favicon.ico:1]: only use pages dir to avoid confusing

File: apps/web/app/page.module.css
Comment (exneval) [inline apps/web/app/page.module.css:1]: only use pages dir to avoid confusing

File: apps/web/app/page.tsx
Comment (exneval) [inline apps/web/app/page.tsx:1]: only use pages dir to avoid confusing

**Review by exneval — CHANGES_REQUESTED** [review]

File: apps/blog/pages/about/index.tsx (hunk 2 above)
Comment (alvinisasi) [reply apps/blog/pages/about/index.tsx]: fixed on #3

File: apps/blog/pages/about/index.tsx (hunk 1 above)
Comment (alvinisasi) [reply apps/blog/pages/about/index.tsx]: fixed on #3

File: apps/blog/pages/about/index.tsx (hunk 3 above)
Comment (alvinisasi) [reply apps/blog/pages/about/index.tsx]: fixed on #2

File: apps/blog/pages/posts/[slug]/index.tsx (hunk 4 above)
Comment (alvinisasi) [reply apps/blog/pages/posts/[slug]/index.tsx]: fixed on #3

File: apps/blog/pages/posts/[slug]/index.tsx (hunk 5 above)
Comment (alvinisasi) [reply apps/blog/pages/posts/[slug]/index.tsx]: fixed on #3

File: apps/web/app/layout.tsx
Comment (alvinisasi) [reply apps/web/app/layout.tsx:1]: fixed on #5

File: apps/web/app/globals.css
Comment (alvinisasi) [reply apps/web/app/globals.css:1]: fixed on #5

File: apps/web/app/favicon.ico
Comment (alvinisasi) [reply apps/web/app/favicon.ico:1]: fixed on #5

File: apps/web/app/page.module.css
Comment (alvinisasi) [reply apps/web/app/page.module.css:1]: fixed on #5

File: apps/web/app/page.tsx
Comment (alvinisasi) [reply apps/web/app/page.tsx:1]: fixed on #5

**Review by exneval — APPROVED** [review]
LGTM

**Pull request merged**